packages/typegpu/src/std/texture.ts

```diff
@@ -118,3 +134,282 @@ export const textureSample: TextureSampleOverload = createDualImpl(
     };
   },
 );
+
+type TextureSampleLevelOverload = {
+  <T extends TgpuSampledTexture<'2d'>>(
+    texture: T,
+    sampler: TgpuSampler,
+    coords: v2f,
+    level: number,
+  ): v4f;
+  <T extends TgpuSampledTexture<'2d'>>(
+    texture: T,
+    sampler: TgpuSampler,
+    coords: v2f,
+    level: number,
+    offset: v2i,
+  ): v4f;
```

Comment:
```suggestion
  <T extends TgpuSampledTexture<'2d'>>(
    texture: T,
    sampler: TgpuSampler,
    coords: v2f,
    level: number,
    offset?: v2i,
  ): v4f;
```
Could we simplify it like this? (same goes for other overloads in this file)